### ISSUE TITLE
Refined runtime configuration

### ISSUE THREAD

**Issue by strycore** [lutris/lutris#5243]
Right now we use the Ubuntu 18.04 based Lutris runtime. The name of this runtime is hardcoded in our source code. We consider moving away from this runtime and upgrading to something more recent. Doing so might introduce some breakage, some game may still depend on the Ubuntu 18.04 runtime and will break when using something newer.

Instead of hardcoding the runtime names, we can depend on the Lutris API to provide a list of runtimes. This is already the case but we need a way to differentiate real runtimes (like Ubuntu 18 or the upcoming Debian 11) from utilities like innoextract and 7z. 

We would remove the "disable Lutris runtime" and "prefer system libraries" options and replace them with an UI that lets users select individual runtime components and also drag them up or down. "System libraries" should be part of the runtime options but should be always checked and greyed out. It should be possible to move it up or down.  

In addition to introducing a new runtime while still being able to keep the old one around, we can also re-introduce support for the Steam runtime, if detected. We can also support custom runtime which are user specified directories. 

This would also be the right place to experiment with Flatpak/snap/pressure-vessel based runtimes. We don't have them yet but basically they would replace our runtime system entirely. We can have one for Java games, one for 64bit native games, one for 32bit games, etc...

**Comment by danieljohnson2** [comment]
I do not understand this stuff about 'moving up and down'; sounds like some kind of priority order of runtimes, but what would use that?

It seems to me that any given game would have a specific runtime, like Wine games have a specific Wine version.

**Comment by strycore** [comment]
Yes, this would be to change the order of paths in resulting LD_LIBRARY_PATH variable. We already have this with "Prefer system libraries" which puts system paths in front of the Lutris runtime paths. 

In the past, I have considered game specific runtimes which are dynamically built based on the game dependencies, but that would be very difficult to maintain. Like most Wine games run with the latest Wine-GE, most native games will work on the default runtime.
